sympy/solvers/tests/test_solvers.py

```diff
@@ -2113,3 +2113,8 @@ def test_issue_17452():
     assert solve((7**x)**x + pi, x) == [-sqrt(log(pi) + I*pi)/sqrt(log(7)),
                                         sqrt(log(pi) + I*pi)/sqrt(log(7))]
     assert solve(x**(x/11) + pi/11, x) == [exp(LambertW(-11*log(11) + 11*log(pi) + 11*I*pi))]
+
+
+def test_issue_17650():
+    x = Symbol('x', real=True)
+    assert solve(abs((abs(x**2-1)-x))-x) == [1, -1 + sqrt(2), 1 + sqrt(2)]
```

Comment:
Please put spaces around `-`, readability is too low.

Reply:
Does `solve(abs((abs(x**2-1)-x))-x)` always return canonical results?

Reply:
What do you mean by canonical results?